### DISCUSSION TITLE
Is it possible to obtain job outputs from another workflow?

### DISCUSSION THREAD

**Discussion by fabiocarneiro** [community/community#33722] [Actions] [answered]
I am triggering a deploy workflow based on the workflow_run, and I would like to obtain a tag that is generated by https://github.com/marketplace/actions/docker-metadata-action. I am passing the step output as the job output, but I have no idea on how to or if it is possible to obtain that value as part of the new workflow.

I did find [this guide](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run) that talks about downloading artifacts from the API, but not sure the same would be available for step/job outputs.

**Comment by airtower-luna** [comment, marked as answer]
A job output is only available within the workflow. The artifact guide you found is probably the way to go here (upload the data you want as an artifact, download in the next workflow).

Another options would be to see if it makes sense to create another job within the workflow instead of a separate workflow, but whether that makes sense depends on your exact context.

**Reply by fabiocarneiro** [reply]
It's kind of a bummer that this is not possible. The solution with making a simple string a artifact feels like overkill.

I initially had it as a single workflow but decided to split both because the documentation suggests to run the deploy as a separate workflow (maybe with higher privileges) and because I wanted to add a filter so the deploy job would not run in all builds.

**Reply by airtower-luna** [reply]
> I initially had it as a single workflow but decided to split both because the documentation suggests to run the deploy as a separate workflow (maybe with higher privileges)

If you mean adjusting token permissions, you [can set those per-job, too](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idpermissions). Otherwise it depends on the kind of privileges you mean.

> because I wanted to add a filter so the deploy job would not run in all builds.

You should be able to do that with a [job-level `if`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idif), the syntax is just different from `on` filters. :sweat_smile:

**Reply by fabiocarneiro** [reply]
That helps a lot, but as it was correctly outlined [here](https://stackoverflow.com/questions/58139406/only-run-job-on-specific-branch-with-github-actions) and [here](https://github.com/orgs/community/discussions/26349), declaring multiple branches can be quite verbose and when using contains there seems to be a potential for false positives.

Ideally, the job would also support `on` conditions. :(
or it would be easier to pass data to the deploy workflow

I don't think its such a rare requirement to only run deploy jobs from specific branches.

**Reply by airtower-luna** [reply]
If you mean the [`contains()` function](https://docs.github.com/en/actions/learn-github-actions/expressions#contains), it should be precise if you search in an array rather than a string. I'm thinking of something like this:

```
if: contains(['main', 'stable'], github.ref_name)
```

That way it should return `true` only if the string is an item of the array (one of the listed strings). I have no idea why the examples you found bother with substrings instead. :sweat_smile:
